noir-r1cs/src/sparse_matrix.rs

```diff
@@ -156,16 +159,40 @@ impl Mul<&[FieldElement]> for HydratedSparseMatrix<'_> {
             rhs.len(),
             "Vector length does not match number of columns."
         );
+
         let mut result = vec![FieldElement::zero(); self.matrix.num_rows];
-        for ((i, j), value) in self.iter() {
-            result[i] += value * rhs[j];
-        }
+
+        (0..self.matrix.num_rows)
+            .into_par_iter()
+            .map(|i| {
+                self.iter_row(i)
+                    .fold(FieldElement::zero(), |sum, (j, value)| sum + value * rhs[j])
+            })
+            .collect_into_vec(&mut result);
```

Comment:
`collect_into_vec` clears the vec, so there's no need to initialize it. You can just allocate with `Vec::with_capacity`

Reply:
Alternatively, it should also be possible to do this with something like:

`result.par_iter_mut().enumerate().for_each(|(i, x)| *x = self.iter_row(i)... );`